src/frontend/src/binder/expr/function/mod.rs

```diff
@@ -166,9 +167,13 @@ impl Binder {
                     InputRef::new(i, DataType::List(Box::new(expr.return_type()))).into()
                 })
                 .collect_vec();
-            let scalar_func_expr = if let Ok(schema) = self.first_valid_schema()
-                && let Some(func) = schema.get_function_by_name_inputs(&func_name, &mut array_args)
-            {
+            let schema_path = self.bind_schema_path(schema_name.as_deref());
+            let scalar_func_expr = if let Ok((func, _)) = self.catalog.get_function_by_name_inputs(
```

Comment:
I'm confused. When should we use  `schema.get_xxx` and `first_valid_schema`? (Can we delete or make private them?)

Reply:
`schema.get_xxx` and `first_valid_schema` should only be used to resolve schema and check duplicate when creating a new function. +1 to make them private.
Btw there is still one issue in `handle_show_create_object` that needs further modifications, but it doesn't seem to be a major problem. NTFS.